examples/tutorials/analysis-time/Variability_estimation.py

```diff
@@ -187,39 +189,29 @@
     t=time, x=flux.flatten(), sigma=flux_err.flatten(), fitness="measures"
 )
 
-######################################################################
-# We can visualize the difference between the original lightcurve and the rebin with bayesian blocks
-
-bayesian_flux = []
-for tmin, tmax in zip(bayesian_edges[:-1], bayesian_edges[1:]):
-    mask = (time >= tmin) & (time <= tmax)
-    bayesian_flux.append(
-        np.average(
-            flux.flatten()[mask], weights=1 / (flux_err.flatten()[mask] ** 2)
-        ).value
-    )
-
-xerr = np.diff(bayesian_edges) / 2
-bayesian_x = bayesian_edges[:-1] + xerr
-
-fig, ax = plt.subplots(
-    figsize=(8, 6),
-    gridspec_kw={"left": 0.16, "bottom": 0.2, "top": 0.98, "right": 0.98},
-)
-plt.plot(time, flux.flatten(), marker="+", linestyle="", color="teal")
-plt.errorbar(
-    bayesian_x,
-    bayesian_flux,
-    xerr=np.diff(bayesian_edges) / 2,
-    linestyle="",
-    color="orange",
-)
-plt.ylim(bottom=0)
-plt.show()
-
 ######################################################################
 # The result giving a significance estimation for variability in the lightcurve is the number of *change points*,
 # i.e. the number of internal bin edges: if at least one change point is identified by the algorithm, there is significant variability.
 
 ncp = len(bayesian_edges) - 2
 print(ncp)
+
+######################################################################
+# We can rebin the lightcurve to compute the one expected with bayesian edges
+# First, we adjust the first and last bins of the bayesian_edges to coincide
+# with the original light curve start and end points.
+
+# create a new axis
+axis_original = lc_1d.geom.axes["time"]
+bayesian_edges[0] = axis_original.time_edges[0].value
+bayesian_edges[-1] = axis_original.time_edges[-1].value
+edges = Time(bayesian_edges, format="mjd", scale=axis_original.reference_time.scale)
```

Comment:
Isn't it possible to pass `bayesian_blocks` a `Time` object directly?

Reply:
No! Maybe it is worth filing an issue in astropy.